accounts/abi/abigen/bindv2.go

```diff
@@ -0,0 +1,376 @@
+// bindings.
+func (b *binder) BindStructType(typ abi.Type) {
+	bindStructType(typ, b.structs)
+}
+
+// contractBinder holds state for binding of a single contract. It is a type
+// registry for compiling maps of identifiers that will be emitted in generated
+// bindings.
+type contractBinder struct {
+	binder *binder
+
+	// all maps are keyed by the original (non-normalized) name of the symbol in question
+	// from the provided ABI definition.
+	calls            map[string]*tmplMethod
+	events           map[string]*tmplEvent
+	errors           map[string]*tmplError
+	callIdentifiers  map[string]bool
+	eventIdentifiers map[string]bool
+	errorIdentifiers map[string]bool
+}
+
+func newContractBinder(binder *binder) *contractBinder {
+	return &contractBinder{
+		binder,
+		make(map[string]*tmplMethod),
+		make(map[string]*tmplEvent),
+		make(map[string]*tmplError),
+		make(map[string]bool),
+		make(map[string]bool),
+		make(map[string]bool),
+	}
+}
+
+// registerIdentifier applies alias renaming, name normalization (conversion
+// from snake to camel-case), and registers the normalized name in the specified identifier map.
+// It returns an error if the normalized name already exists in the map.
+func (cb *contractBinder) registerIdentifier(identifiers map[string]bool, original string) (normalized string, err error) {
+	normalized = abi.ToCamelCase(alias(cb.binder.aliases, original))
+
+	// Name shouldn't start with a digit. It will make the generated code invalid.
+	if len(normalized) > 0 && unicode.IsDigit(rune(normalized[0])) {
+		normalized = fmt.Sprintf("E%s", normalized)
+		normalized = abi.ResolveNameConflict(normalized, func(name string) bool {
+			_, ok := identifiers[name]
+			return ok
+		})
+	}
+	if _, ok := identifiers[normalized]; ok {
+		return "", fmt.Errorf("duplicate symbol '%s'", normalized)
+	}
+	identifiers[normalized] = true
+	return normalized, nil
+}
+
+// bindMethod registers a method to be emitted in the bindings. The name, inputs
+// and outputs are normalized. If any inputs are struct-type their structs are
+// registered to be emitted in the bindings. Any methods that return more than
+// one output have their results gathered into a struct.
+func (cb *contractBinder) bindMethod(original abi.Method) error {
+	normalized := original
+	normalizedName, err := cb.registerIdentifier(cb.callIdentifiers, original.Name)
+	if err != nil {
+		return err
+	}
+	normalized.Name = normalizedName
+
+	normalized.Inputs = normalizeArgs(original.Inputs)
+	for _, input := range normalized.Inputs {
+		if hasStruct(input.Type) {
+			cb.binder.BindStructType(input.Type)
+		}
+	}
+	normalized.Outputs = normalizeArgs(original.Outputs)
+	for j, output := range normalized.Outputs {
+		if output.Name != "" {
+			normalized.Outputs[j].Name = abi.ToCamelCase(output.Name)
+		}
+		if hasStruct(output.Type) {
+			cb.binder.BindStructType(output.Type)
+		}
+	}
+	isStructured := structured(original.Outputs)
```

Comment:
This line is not relevant anymore. With a normalized output arguments, `isStructured` is always true if the length > 1